globus_action_provider_tools/flask/helpers.py

```diff
@@ -0,0 +1,48 @@
+from enum import Enum
+from typing import Iterable, List, Set
+
+from flask import request
+
+
+def parse_query_args(
+    arg_name: str, default_value: str = "", valid_vals: List[str] = None
+) -> Set[str]:
+    """
+    Helper function to parse a query arg "arg_name" and return a validated
+    (according to the values supplied in "valid_vals"), usable set of
+    values (which defaults to "default_value").
+
+    Note that this helper operates within a Flask request context to inspect
+    query parameters.
+    """
+    param_val = request.args.get(arg_name, default_value)
+
+    # A query param name with no value returns empty string even with default value
+    if param_val == "":
+        param_val = default_value
+
+    # Split in case there's a comma seperated query param value
+    param_vals = set(param_val.split(","))
+
+    # Remove invalid data from query params
+    if valid_vals is not None:
```

Comment:
If it weren't for the `lower` thing, which is probably a good idea, this could probably just be an intersect operation or some such. That'd be pretty elegant...